File: core/src/main/kotlin/org/bitlap/core/mdm/format/DataType.kt
/* Copyright (c) 2022 bitlap.org */
package org.bitlap.core.mdm.format

import java.io.Serializable

abstract class DataType : Serializable {

    abstract val name: String
    abstract val idx: Int

    override fun toString(): String {
        return "DataType(name='$name', idx=$idx)"
    }
}

File: bitlap-core-s/src/main/scala/org/bitlap/core/mdm/format/DataType.scala
/** Copyright (C) 2023 bitlap.org .
 */
package org.bitlap.core.mdm.format

import java.io.Serializable

abstract class DataType extends Serializable {

  val name: String
  val idx: Int

  override def toString: String = {
    return s"$name#$idx"
  }
}
